Microsoft.Toolkit.Uwp.UI.Controls/AdvancedCollectionViewSource/AdvancedCollectionViewSource.cs

```diff
@@ -0,0 +1,560 @@
+        public object this[int index]
+        {
+            get { return _view[index]; }
+            set { _view[index] = value; }
+        }
+
+        /// <summary>
+        /// Occurs when the vector changes.
+        /// </summary>
+        public event VectorChangedEventHandler<object> VectorChanged;
+
+        /// <summary>
+        /// Move current index to item
+        /// </summary>
+        /// <param name="item">item</param>
+        /// <returns>success of operation</returns>
+        public bool MoveCurrentTo(object item) => item == CurrentItem || MoveCurrentToIndex(IndexOf(item));
+
+        private bool MoveCurrentToIndex(int i)
+        {
+            if (i < -1 || i >= _view.Count)
+            {
+                return false;
+            }
+
+            if (i == _index)
+            {
+                return false;
+            }
+
+            var e = new CurrentChangingEventArgs();
+            OnCurrentChanging(e);
+            if (e.Cancel)
+            {
+                return false;
+            }
+
+            _index = i;
+            OnCurrentChanged(null);
+            return true;
+        }
+
+        /// <summary>
+        /// Moves selected item to position
+        /// </summary>
+        /// <param name="index">index</param>
+        /// <returns>success of operation</returns>
+        public bool MoveCurrentToPosition(int index) => MoveCurrentToIndex(index);
+
+        /// <summary>
+        /// Move current item to first item
+        /// </summary>
+        /// <returns>success of operation</returns>
+        public bool MoveCurrentToFirst() => MoveCurrentToIndex(0);
+
+        /// <summary>
+        /// Move current item to last item
+        /// </summary>
+        /// <returns>success of operation</returns>
+        public bool MoveCurrentToLast() => MoveCurrentToIndex(_view.Count - 1);
+
+        /// <summary>
+        /// Move current item to next item
+        /// </summary>
+        /// <returns>success of operation</returns>
+        public bool MoveCurrentToNext() => MoveCurrentToIndex(_index + 1);
+
+        /// <summary>
+        /// Move current item to previous item
+        /// </summary>
+        /// <returns>success of operation</returns>
+        public bool MoveCurrentToPrevious() => MoveCurrentToIndex(_index - 1);
+
+        /// <summary>
+        /// Load more items from the source
+        /// </summary>
+        /// <param name="count">number of items to load</param>
+        /// <returns>Async operation of LoadMoreItemsResult</returns>
+        /// <exception cref="NotImplementedException">Not implemented yet...</exception>
+        public IAsyncOperation<LoadMoreItemsResult> LoadMoreItemsAsync(uint count)
+        {
+            throw new NotImplementedException("todo...");
```

Comment:
Needs to be completed. I assume it's just a matter of checking if the source implements ISupportIncrementalLoading and call LoadMoreItemsAsync on that ?